tests/diff.js

```diff
@@ -67,4 +67,88 @@ describe( 'diff', () => {
 		diff( getLongText( 10 ), getLongText( 1990 ) );
 		testUtils.sinon.assert.called( fastDiffSpy );
 	} );
+
+	describe( 'with multi-byte unicode', () => {
+		describe( 'simple emoji - single unicode code point', () => {
+			// 🙂 = '\ud83d\ude42' = 2 chars
+			const emojiLength = '🙂'.length;
+			const emojiDiffInsert = new Array( emojiLength ).fill( 'insert' );
+			const emojiDiffEqual = new Array( emojiLength ).fill( 'equal' );
+			const emojiDiffDelete = new Array( emojiLength ).fill( 'delete' );
+
+			it( 'should properly handle emoji insertion', () => {
+				expect( diff( 'abc', 'ab🙂c' ) ).to.deep.equals( [ 'equal', 'equal', ...emojiDiffInsert, 'equal' ] );
+			} );
+
+			it( 'should properly handle emoji insertion on the end', () => {
+				expect( diff( 'abc', 'abc🙂' ) ).to.deep.equals( [ 'equal', 'equal', 'equal', ...emojiDiffInsert ] );
+			} );
+
+			it( 'should properly handle appending to string containing emoji', () => {
+				expect( diff( 'abc🙂', 'abc🙂d' ) ).to.deep.equals( [ 'equal', 'equal', 'equal', ...emojiDiffEqual, 'insert' ] );
+			} );
+
+			it( 'should properly handle insertion to string containing emoji', () => {
+				expect( diff( 'ab🙂cd', 'ab🙂cde' ) ).to.deep.equals( [ 'equal', 'equal', ...emojiDiffEqual, 'equal', 'equal', 'insert' ] );
+			} );
+
+			it( 'should properly remove emoji', () => {
+				expect( diff( 'a🙂b', 'ab' ) ).to.deep.equals( [ 'equal', ...emojiDiffDelete, 'equal' ] );
+			} );
+
+			it( 'should properly replace emoji', () => {
```

Comment:
I'm missing a case with replacing one emoji with another one.

Reply:
Another test I miss is for several emojis one after another. It creates some additional cases. I don't expect those to ever break, unless we'll start doing some weird code optimizations in diff/fastDiff, so those tests may not be essential, but it's easy to have them just in case.